src/utils/date.ts

```diff
@@ -10,4 +11,127 @@ export function isDate(dateString?: ValueType) {
   }
 
   return !isNaN(new Date(dateString).getDate());
-}
\ No newline at end of file
+}
+
+export enum DATE_FORMATS {
+  TIME = "h:mm A",
+  SHORT = "MMM DD",
+  LONG = "h:mm A, MMM DD, YYYY",
+  TIME_RANGE = "YYYY-MM-DD HH:mm"
+}
+
+type formatDateOptions = {
+  suffix?: string;
+  asTimestamp?: boolean;
+  stringFormat?: DATE_FORMATS;
+};
+
+/**
+ * Format date into local time and apply the string formatting if given
+ * else the date object is parsed into local time
+ *
+ * @param date date which needs to be parsed or formatted
+ * @param stringFormat string of tokens to format the date, check https://day.js.org/docs/en/display/format#docsNav
+ * @returns string if stringFormat is passed else a date object
+ */
+export const formatDate = (
+  date: string | Date,
+  { asTimestamp, stringFormat, suffix }: formatDateOptions = {}
+) => {
+  const localDate = dayjs(date).local();
+  if (asTimestamp) return localDate.valueOf().toString();
+  const formattedDate = stringFormat
+    ? localDate.format(stringFormat)
+    : localDate;
+  return suffix ? `${formattedDate}${suffix}` : formattedDate;
+};
+
+/**
+ * Parse the given date relative to the user locale
+ *
+ * @param date
+ * @param fromNow give relative date from now
+ * @returns relative date as string
+ */
+export const relativeDateTime = (date: string | Date, fromNow = false) => {
```

Comment:
Why would a relative date be anything from now ?

Reply:
The function is not exactly giving only the relative date. The way it was originally written , it either gives the duration passed from now for the "date" passed in the parameter (if "fromNow" is set to true). If "fromNow" is not specified, it gives duration only if the "date" parameter passed is the same day as today. Else it gives the date-time format as specified but with respect to the user local. It is being used in a lot of places, so I assumed the working of the function is right.

Let me know if we need to make changes to this function.

Reply:
We don't want to have inconsistent dates, relalativeDate should always return the duration from now